dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/CallbackServiceCodec.java

```diff
@@ -201,11 +207,16 @@ private Object referOrDestroyCallbackService(Channel channel, URL url, Class<?>
         if (isRefer) {
             if (proxy == null) {
                 URL referurl = URL.valueOf("callback://" + url.getAddress() + "/" + clazz.getName() + "?" + INTERFACE_KEY + "=" + clazz.getName());
-                referurl = referurl.addParametersIfAbsent(url.getParameters()).removeParameter(METHODS_KEY);
+                referurl = referurl.addParametersIfAbsent(url.getParameters()).removeParameter(METHODS_KEY).addParameter(SIDE_KEY, CONSUMER_SIDE);
                 if (!isInstancesOverLimit(channel, referurl, clazz.getName(), instid, true)) {
                     url.getOrDefaultApplicationModel().getDefaultModule().getServiceRepository().registerService(clazz);
                     @SuppressWarnings("rawtypes")
                     Invoker<?> invoker = new ChannelWrappedInvoker(clazz, channel, referurl, String.valueOf(instid));
+
+                    FilterChainBuilder builder = getFilterChainBuilder(url);
+                    invoker = builder.buildInvokerChain(invoker, REFERENCE_FILTER_KEY, CommonConstants.CONSUMER);
+                    invoker = builder.buildInvokerChain(invoker, REFERENCE_FILTER_KEY, CommonConstants.CALLBACK);
```

Comment:
By adding the above two lines, both ConsumerContextFilter and CallbackConsumerContextFilter will be enabled. 

What's the meaning of enable both these two filters as they have exact the same effect?